kube-core/src/conversion/types.rs

```diff
@@ -0,0 +1,178 @@
+#[derive(Serialize, Deserialize)]
+pub struct ConversionReview {
+    /// Contains the API version and type of the request.
+    #[serde(flatten)]
+    pub types: TypeMeta,
+    /// Contains conversion request.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub request: Option<ConversionRequest>,
+    /// Contains conversion response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub response: Option<ConversionResponse>,
+}
+
+/// Part of ConversionReview which is set on input (i.e. generated by apiserver)
+#[derive(Serialize, Deserialize)]
+pub struct ConversionRequest {
+    /// Copied from the corresponding consructing [`ConversionReview`].
+    /// This field is not part of the Kubernetes API, it's consumed only by `kube`.
+    #[serde(skip)]
+    pub types: Option<TypeMeta>,
+    /// Random uid uniquely identifying this conversion call
+    pub uid: String,
+    /// The API group and version the objects should be converted to
+    #[serde(rename = "desiredAPIVersion")]
+    pub desired_api_version: String,
+    /// The list of objects to convert.
+    /// May contain one or more objects, in one or more versions.
+    // This field used raw Value instead of Object/DynamicObject to simplify
+    // further downcasting.
+    pub objects: Vec<serde_json::Value>,
+}
+
+impl ConversionRequest {
+    /// Extracts request from the [`ConversionReview`].
+    /// Fails if review has missing request.
+    pub fn from_review(review: ConversionReview) -> Result<Self, ConvertConversionReviewError> {
+        match review.request {
+            Some(mut req) => {
+                req.types = Some(review.types);
+                Ok(req)
+            }
+            None => Err(ConvertConversionReviewError),
+        }
+    }
+}
+
+impl TryFrom<ConversionReview> for ConversionRequest {
+    type Error = ConvertConversionReviewError;
+
+    fn try_from(review: ConversionReview) -> Result<Self, Self::Error> {
+        ConversionRequest::from_review(review)
+    }
+}
+
+/// Part of ConversionReview which is set on output (i.e. generated by conversion webhook)
+#[derive(Serialize, Deserialize)]
+pub struct ConversionResponse {
+    /// Copied from the corresponding consructing [`ConversionRequest`].
+    /// This field is not part of the Kubernetes API, it's consumed only by `kube`.
+    #[serde(skip)]
+    pub types: Option<TypeMeta>,
+    /// Copy of .request.uid
+    pub uid: String,
+    /// Outcome of the conversion operation.
+    /// Success: all objects were successfully converted
+    /// Failure: at least one object could not be converted.
+    /// It is recommended that conversion fails as rare as possible.
+    pub result: Status,
+    /// Converted objects in the same order as in the request. Should be empty
+    /// if conversion failed.
+    #[serde(rename = "convertedObjects")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub converted_objects: Option<Vec<serde_json::Value>>,
```

Comment:
As far as I remember, without this Option example webhook failed on [this](https://github.com/kube-rs/kube-rs/pull/999/files#diff-3ba17590977dafaa6f5ba1ff1d3db64d0531ebfaf3bb23221edb2978fe756273R50) part of input.

Reply:
If that is a legit response that Kubernetes accepts then that's OK, but it does seem like the docs indicate we should be putting in an explicit empty vec in the failure case rather than a null.

Reply:
I changed implementation so that field is typed as `Vec<Value>` (and null is special-cased during parsing).